### ISSUE TITLE
question about gif format

### ISSUE THREAD

**Issue by aHyunjaePark** [moxiecode/moxie#84]
Hi, 

thanks for the awesome open source, it literally helped me a lot :)

I'm trying to make a preview image viewer, before uploading with plupload.

followed this code.

      function showImagePreview( file ) {
 
                var item = $( "<li></li>" ).prependTo( dom.uploads );
                var image = $( new Image() ).appendTo( item );
 
                // Create an instance of the mOxie Image object. This
                // utility object provides several means of reading in
                // and loading image data from various sources.
                // --
                // Wiki: https://github.com/moxiecode/moxie/wiki/Image
                var preloader = new mOxie.Image();
 
                // Define the onload BEFORE you execute the load()
                // command as load() does not execute async.
                preloader.onload = function() {
 
                    // This will scale the image (in memory) before it
                    // tries to render it. This just reduces the amount
                    // of Base64 data that needs to be rendered.
                    preloader.downsize( 300, 300 );
 
                    // Now that the image is preloaded, grab the Base64
                    // encoded data URL. This will show the image
                    // without making an Network request using the
                    // client-side file binary.
                    image.prop( "src", preloader.getAsDataURL() );
 
                    // NOTE: These previews "work" in the FLASH runtime.
                    // But, they look seriously junky-to-the-monkey.
                    // Looks like they are only using like 256 colors.
 
                };
 
                // Calling the .getSource() on the file will return an
                // instance of mOxie.File, which is a unified file
                // wrapper that can be used across the various runtimes.
                // --
                // Wiki: https://github.com/moxiecode/plupload/wiki/File
                preloader.load( file.getSource() );
 
            }

source : 

http://www.bennadel.com/blog/2563-showing-plupload-image-previews-using-base64-encoded-data-urls.htm

In here, this works fine, if the upload file is png, or jpg format.

but it doesn't work, when the format is gif format :(

gif format is not supported? or is the code bug?

in here,

https://github.com/moxiecode/moxie/wiki/Image#getAsDataURL-typeimagejpegquality90-method

it says,

[type="image/jpeg"] String
Mime type of resulting blob. Can either be image/jpeg or image/png

so it means gif is not supported?

I'll be waiting for response 

Thank  you!

**Comment by PhilippMikael** [comment]
I vote for this. Need gif support to embed image thumbnails, going to be uploaded with plupload.

**Comment by jayarjo** [comment]
What do you think we should do with animated gifs?

**Comment by PhilippMikael** [comment]
Good point. The basic version could only render the first frame of the GIF. While I'm using moxie image for embedded preview thumbnails in a Plupload queue, this would be OK for my use case. Although it is possible to scale an animated GIF (including all frames) with software like ImageMagick, I don't now if there is a relatively easy way to accomplish this with pure JavaScript.
